jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/ProbeHandler.java

```diff
@@ -93,6 +89,22 @@ private HTTPGetAction getHTTPGetAction(String getUrl) {
         }
     }
 
+    private String resolveHost(String host) {
+        if (host.equalsIgnoreCase("localhost")) {
+            // No need to set host in case of localhost HTTP request, will default to Pod IP
+            return null;
+        }
+        return host;
+    }
```

Comment:
Why is this necessary?

Reply:
This is done in order to support local probes. If we set the host to null, it is assumed to be pod's IP. This way when a user provides `<getUrl>http://localhost:8080/actuator/health</getUrl>`, it generates a probe like this:
```yaml
        readinessProbe:
          httpGet:
            path: /actuator/health
            port: 8080
            scheme: HTTP
```

Reply:
I see what it does, but local probes were supported before (weren't they?), so why is this necessary now?

Reply:
In order to provide local probes user needed to provide host omitted getUrl like this: `<getUrl>http://:8080/actuator/health</getUrl>` . I thought providing localhost could be a cleaner alternative

Reply:
We should revert this (it won't consider cases such as 127.0.0.1) and it is misleading for users.
The host part of the [probe config](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#http-probes) should be discouraged
> host: Host name to connect to, defaults to the pod IP. You probably want to set "Host" in httpHeaders instead.

Reply:
@rohanKanojia , please update the PR description to match the new behavior:
> Added a check to keep host null in case it matches localhost